std/zip.d

```diff
@@ -823,6 +825,78 @@ final class ZipArchive
     {
         data[i .. i + 8] = nativeToLittleEndian(ul);
     }
+
+    /* ============== for detecting overlaps =============== */
+
+private:
+
+    // defines a segment of the zip file, including start, excluding end
+    struct Segment
+    {
+        uint start;
+        uint end;
+    }
+
+    // removes Segment start .. end from _segs
+    // throws zipException if start .. end is not completely available in _segs;
+    void removeSegment(uint start, uint end) pure @safe
+    {
+        if (_segs.length == 0)
+            throw new ZipException("overlapping data detected");
+
+        uint best = 0;
+        size_t bestPos = 0;
+        foreach (i,seg;_segs)
```

Comment:
No. The segments will represent the parts *not* identified. 

In theory this can indeed lead to an overall performance of O(n^2), when the files are randomly distributed inside the zip file. But in normal (todays) zip archives they are lined up and in this case at most 3 segments are needed (one at the start if it's selfextracting, one between the files and the directory and one at the end of the directory), making this foreach O(1). I think, making this faster could be an improvement, left for later. But for the time being, I'd like to stick to a simple algorithm to avoid bugs.